src/common/mfu_daos.c

```diff
@@ -1313,3 +1330,2195 @@ int mfu_flist_copy_daos(
+            }
+            /* created attribute in HDF5 file with encoded
+             * dataspace for this record extent */
+            memset(attr_name, 64, sizeof(attr_name));
+            memset(number_str, 16, sizeof(number_str));
+            memset(attr_num_str, 16, sizeof(attr_num_str));
+            path_len = snprintf(number_str, 10, "%d",
+                        (int)(*ak_index));
+            if (path_len >= 16) {
+                MFU_LOG(MFU_LOG_ERR, "number_str is too long");
+                rc = 1;
+                goto out;
+            }
+            path_len = snprintf(attr_num_str, 10, "-%d", attr_num);
+            if (path_len >= 16) {
+                MFU_LOG(MFU_LOG_ERR, "attr number str is too long");
+                rc = 1;
+                goto out;
+            }
+            path_len = snprintf(attr_name, 64, "%s", "A-");
+            if (path_len >= 64) {
+                MFU_LOG(MFU_LOG_ERR, "attr name is too long");
+                rc = 1;
+                goto out;
+            }
+            strcat(attr_name, number_str);
+            strcat(attr_name, attr_num_str);
+            printf("\n\nATTR NAME: %s\n\n", attr_name);
+            encode_buf_len = nalloc * sizeof(unsigned char);
+            hdf5->attr_dims[0] = encode_buf_len;
+            hdf5->attr_dspace = H5Screate_simple(1, hdf5->attr_dims,
+                                     NULL);
+            if (hdf5->attr_dspace < 0) {
+                MFU_LOG(MFU_LOG_ERR, "failed to create attr");
+                rc = 1;
+                goto out;
+            }
+            hdf5->selection_attr = H5Acreate2(hdf5->rx_dset,
+                                              attr_name,
+                                              hdf5->rx_dtype,
+                                              hdf5->attr_dspace,
+                                              H5P_DEFAULT,
+                                              H5P_DEFAULT);
+            if (hdf5->selection_attr < 0) {
+                MFU_LOG(MFU_LOG_ERR, "failed to create selection attr");
+                rc = 1;
+                goto out;
+            }   
+            status = H5Awrite(hdf5->selection_attr, hdf5->rx_dtype,
+                              encode_buf);
+            if (status < 0) {
+                MFU_LOG(MFU_LOG_ERR, "failed to write attr");
+                rc = 1;
+                goto out;
+            }
+            status = H5Aclose(hdf5->selection_attr);
+            if (status < 0) {
+                MFU_LOG(MFU_LOG_ERR, "failed to close attr");
+                rc = 1;
+                goto out;
+            }
+            status = H5Sclose(hdf5->rx_memspace);
+            if (status < 0) {
+                MFU_LOG(MFU_LOG_ERR, "failed to close rx_memspace");
+                rc = 1;
+                goto out;
+            }
+            if (encode_buf != NULL) 
+                free(encode_buf);
+            attr_num++;
+        }
+    }
+out:
+    return rc;
+}
```

Comment:
@dsikich In general, I don't think everything is free'ed here on error conditions.
We might consider declaring as many variables as we can at the top of this function and initializing them to NULL. Then, we can easily call mfu_free here on each one of them if rc is not 0.